### PR TITLE
Upgrade Trivy to v0.21.3

### PR DESCRIPTION
**Pull request by PhilipSchmid**
Hi guys,

Inside this PR, I upgraded Trivy to version `v0.21.3`. Harbor was also upgraded to the next patch version (`v2.4.1`).

I think it needs some further testing from your side, especially since the `make test-component` command currently fails (`make test` and `make test-integration` runs just fine):
```bash
vagrant@ubuntu-focal:/vagrant$ make test-component
GOOS=linux GO111MODULE=on CGO_ENABLED=0 go build -o scanner-trivy cmd/scanner-trivy/main.go
docker build --no-cache -t aquasec/harbor-scanner-trivy:dev .
Sending build context to Docker daemon  46.91MB
Step 1/8 : ARG TRIVY_VERSION=0.21.3
Step 2/8 : FROM aquasec/trivy:${TRIVY_VERSION}
 ---> c28c291ad36e
Step 3/8 : ARG TRIVY_VERSION
 ---> Running in 60f7a8ca7a21
Removing intermediate container 60f7a8ca7a21
 ---> 41e569477c1e
Step 4/8 : RUN adduser -u 10000 -D -g '' scanner scanner
 ---> Running in be5452229b99
Removing intermediate container be5452229b99
 ---> 8dd1e1ffcf6d
Step 5/8 : COPY scanner-trivy /home/scanner/bin/scanner-trivy
 ---> 5a9017706aaa
Step 6/8 : ENV TRIVY_VERSION=${TRIVY_VERSION}
 ---> Running in eeb6e3bffad1
Removing intermediate container eeb6e3bffad1
 ---> 0b47b7a24302
Step 7/8 : USER scanner
 ---> Running in 6e07c21fe43d
Removing intermediate container 6e07c21fe43d
 ---> a7a4b0b73931
Step 8/8 : ENTRYPOINT ["/home/scanner/bin/scanner-trivy"]
 ---> Running in 4f39359b6994
Removing intermediate container 4f39359b6994
 ---> 5eaf0bfba27a
Successfully built 5eaf0bfba27a
Successfully tagged aquasec/harbor-scanner-trivy:dev
GO111MODULE=on go test -count=1 -v -tags=component ./test/component/...
=== RUN   TestComponent
2021/12/24 13:00:36 Starting container id: d48e555d5bd2 image: quay.io/testcontainers/ryuk:0.2.3
2021/12/24 13:00:37 Waiting for container id d48e555d5bd2 image: quay.io/testcontainers/ryuk:0.2.3
2021/12/24 13:00:37 Container is ready id: d48e555d5bd2 image: quay.io/testcontainers/ryuk:0.2.3
2021/12/24 13:00:37 Starting container id: a4882660bf98 image: redis:5
2021/12/24 13:00:38 Waiting for container id a4882660bf98 image: redis:5
2021/12/24 13:00:38 Container is ready id: a4882660bf98 image: redis:5
2021/12/24 13:00:43 Starting container id: f655b67ff925 image: registry:2
2021/12/24 13:00:44 Waiting for container id f655b67ff925 image: registry:2
2021/12/24 13:00:45 Container is ready id: f655b67ff925 image: registry:2
2021/12/24 13:00:45 Starting container id: a62d71d0f3b9 image: aquasec/harbor-scanner-trivy:dev
2021/12/24 13:00:45 Waiting for container id a62d71d0f3b9 image: aquasec/harbor-scanner-trivy:dev
2021/12/24 13:00:54 Container is ready id: a62d71d0f3b9 image: aquasec/harbor-scanner-trivy:dev
=== RUN   TestComponent/Should_scan_alpine:3.14
    component_test.go:180:
        	Error Trace:	component_test.go:180
        	Error:      	Received unexpected error:
        	            	invalid response status: 500 500 Internal Server Error
        	Test:       	TestComponent/Should_scan_alpine:3.14
=== CONT  TestComponent
    component_test.go:131: adatper logs
        [{"built_at":"unknown","commit":"none","level":"info","msg":"Starting harbor-scanner-trivy","time":"2021-12-24T13:00:45Z","version":"dev"}
         {"level":"debug","msg":"Current process","pid":1,"time":"2021-12-24T13:00:45Z"}
         {"gid":10000,"home_dir":"/home/scanner","level":"debug","msg":"Current user","time":"2021-12-24T13:00:45Z","uid":10000}
         {"level":"warning","msg":"trivy cache dir does not exist","path":"/home/scanner/.cache/trivy","time":"2021-12-24T13:00:45Z"}
         {"level":"debug","msg":"Creating trivy cache dir","path":"/home/scanner/.cache/trivy","time":"2021-12-24T13:00:45Z"}
         {"level":"debug","mode":"dgrwxr-xr-x","msg":"trivy cache dir permissions","time":"2021-12-24T13:00:45Z"}
         {"level":"warning","msg":"trivy reports dir does not exist","path":"/home/scanner/.cache/reports","time":"2021-12-24T13:00:45Z"}
         {"level":"debug","msg":"Creating trivy reports dir","path":"/home/scanner/.cache/reports","time":"2021-12-24T13:00:45Z"}
         {"level":"debug","mode":"dgrwxr-xr-x","msg":"trivy reports dir permissions","time":"2021-12-24T13:00:45Z"}
         {"level":"trace","msg":"Connecting to Redis","time":"2021-12-24T13:00:45Z","url":"redis://redis:6379"}
         ERROR: write_concurrency_controls_max_concurrency - dial tcp: lookup redis on 127.0.0.11:53: no such host
         {"addr":":8080","level":"warning","msg":"Starting API server without TLS","time":"2021-12-24T13:00:53Z"}
         {"level":"trace","msg":"Connecting to Redis","time":"2021-12-24T13:00:53Z","url":"redis://redis:6379"}
         {"level":"trace","msg":"Connecting to Redis","time":"2021-12-24T13:00:53Z","url":"redis://redis:6379"}
         {"level":"trace","msg":"Connecting to Redis","time":"2021-12-24T13:00:53Z","url":"redis://redis:6379"}
         {"level":"trace","msg":"Connecting to Redis","time":"2021-12-24T13:00:53Z","url":"redis://redis:6379"}
         {"level":"trace","msg":"Connecting to Redis","time":"2021-12-24T13:00:54Z","url":"redis://redis:6379"}
         {"level":"trace","msg":"172.20.0.1:34184 - HTTP/1.1 POST /api/v1/scan","time":"2021-12-24T13:00:58Z"}
         {"level":"debug","msg":"Enqueueing scan job","time":"2021-12-24T13:00:58Z"}
        ]
--- FAIL: TestComponent (51.08s)
    --- FAIL: TestComponent/Should_scan_alpine:3.14 (15.93s)
FAIL
FAIL	github.com/aquasecurity/harbor-scanner-trivy/test/component	51.100s
?   	github.com/aquasecurity/harbor-scanner-trivy/test/component/docker	[no test files]
?   	github.com/aquasecurity/harbor-scanner-trivy/test/component/scanner	[no test files]
FAIL
make: *** [Makefile:17: test-component] Error 1
```

Also interesting was, that a manual testing of some sample images (`alpine:3.14` and `elasticsearch:6.8.14`) worked like a charm:
 
image
image
image

I think it's a more a failure in the `test-component` setup than a broken setup because of the new Trivy / Harbor version.

BTW, with this new Trivy version, even the log4shell CVEs were now detected and directly reported to Harbor:

image

Thanks & regards,
Philip

### CODE REVIEW

File: test/component/component_test.go [hunk 1]
```diff
@@ -69,7 +69,7 @@ func TestComponent(t *testing.T) {
 
 	redisC, err := dp.CreateContainer(ctx,
 		tc.ContainerRequest{
-			Name:       "redis",
+			Name:       "redis-comp-test",
```
Comment (PhilipSchmid) [inline test/component/component_test.go:72]: Changed to something unique because other ways there's a container name conflict with the Harbor Docker Compose setup.

File: Dockerfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # That's the only place where you're supposed to specify version of Trivy.
-ARG TRIVY_VERSION=0.20.1
+ARG TRIVY_VERSION=0.21.3
```
Comment (danielpacak) [inline Dockerfile:2]: In #215 we bump it to 0.22.0 which is the latest

File: README.md [hunk 3]
```diff
@@ -39,6 +39,7 @@ The following matrix indicates the version of Trivy and Trivy adapter installed
 
 | Harbor          | Trivy Adapter | Trivy           |
 |-----------------|---------------|-----------------|
+| [harbor v2.4.1] | v0.25.0       | [trivy v0.21.3] |
```
Comment (danielpacak) [inline README.md:42]: Usually we update the compatibility matrix once the adapter service is released and version is bumped in the https://github.com/goharbor/harbor repo.

File: helm/harbor-scanner-trivy/values.yaml [hunk 4]
```diff
@@ -4,7 +4,7 @@ fullnameOverride: ""
 image:
   registry: docker.io
   repository: aquasec/harbor-scanner-trivy
-  tag: 0.24.0
+  tag: 0.25.0
```
Comment (danielpacak) [inline helm/harbor-scanner-trivy/values.yaml:7]: This is out of scope. We do release prepare patches to bump up Helm chart and image tags

File: test/component/component_test.go (hunk 1 above)
Comment (danielpacak) [reply test/component/component_test.go:72]: Usually we solve one problem or do one enhancement at a time. If name of the container is an issue we should have a dedicated PR to proceed.

**Review by danielpacak — CHANGES_REQUESTED** [review]
Thank you for working on this patch @PhilipSchmid I'm not sure how to proceed with this one because we already have #215 , that does what it claims. On the other hand, your PR is touching on multiple things, which is not how we manage contributions. I'd close this PR and proceed with merging #215 if you don't mind. WDYT?

**Comment by danielpacak** [comment]
Closing as we have merged #215
